datafusion/sqllogictest/test_files/copy.slt

```diff
@@ -255,31 +257,32 @@ select * from validate_struct_with_array;
 query IT
 COPY source_table
 TO 'test_files/scratch/copy/table_with_options/'
-(format parquet,
-'parquet.compression' snappy,
-'parquet.compression::col1' 'zstd(5)',
-'parquet.compression::col2' snappy,
-'parquet.max_row_group_size' 12345,
-'parquet.data_pagesize_limit' 1234,
-'parquet.write_batch_size' 1234,
-'parquet.writer_version' 2.0,
-'parquet.dictionary_page_size_limit' 123,
-'parquet.created_by' 'DF copy.slt',
-'parquet.column_index_truncate_length' 123,
-'parquet.data_page_row_count_limit' 1234,
-'parquet.bloom_filter_enabled' true,
-'parquet.bloom_filter_enabled::col1' false,
-'parquet.bloom_filter_fpp::col2' 0.456,
-'parquet.bloom_filter_ndv::col2' 456,
-'parquet.encoding' plain,
-'parquet.encoding::col1' DELTA_BINARY_PACKED,
-'parquet.dictionary_enabled::col2' true,
-'parquet.dictionary_enabled' false,
-'parquet.statistics_enabled' page,
-'parquet.statistics_enabled::col2' none,
-'parquet.max_statistics_size' 123,
-'parquet.bloom_filter_fpp' 0.001,
-'parquet.bloom_filter_ndv' 100
+STORED AS PARQUET
+OPTIONS (
+'format.compression' snappy,
```

Comment:
It would be awesome if we could avoid having to repeat out `format.` for each option

Note this change may conflict with the changes from @tinfoil-knight  in https://github.com/apache/arrow-datafusion/pull/9594

Reply:
The challenge here is an option may refer to a non-format thing too. Think about something like:
```
STORED AS PARQUET
OPTIONS (
  format.foo bar
  format.fizz buzz
  credentials.username admin
)
```

With the aligned/unified syntax here, all `format` entries refer to `parquet` and there is no repetition of `parquet`. But we repeat `format` instead, and this prefix is necessary as it separates format-related and non-format-related options.

As a next step, a consistent and generalizable way to remove repetitive patterns would be a syntax like:
```
STORED AS PARQUET
OPTIONS format (
  foo bar
  hey ho
)
OPTIONS other.prefix (
  fizz buzz
  paul atreides
)
OPTIONS (
  credentials.username admin
)
```
where multiple `OPTIONS` are allowable, each with a possible prefix to avoid repetition. In case of an option is specified multiple times with different values, we'd generate an error. The above would be equivalent to
```
STORED AS PARQUET
OPTIONS (
  format.foo bar
  format.hey ho
  other.prefix.fizz buzz
  other.prefix.paul atreides
  credentials.username admin
)
```
where the user is explicitly repeating prefixes.

Reply:
Got it -- I agree that being able to explicitly namespace out the options is valuable to disambiguate them

I think as a user it would be really nice to not have to worry about such collisions unless i actually had to disambiguate them

Like in my mind, this statement isn't ambiguous

```sql
STORED AS PARQUET
OPTIONS (
  compression snappy
)
```

Because the only possible "compression" option is on the format.

I think the way @tinfoil-knight handles this in https://github.com/apache/arrow-datafusion/pull/9594 is to internally try and resolve `compression` as though it were `format.compression` (or `parquet.compression`) if the `compression` configuration isn't valid. 

Even something like this is not ambiguous (though I realize actually implementing this may not be feasible without much more work):

```sql
STORED AS PARQUET
LOCATION 's3://....'
OPTIONS (
  compression snappy
  access_key paul
  secrete_key atreides
)
```



> As a next step, a consistent and generalizable way to remove repetitive patterns would be a syntax like:

That is definitely neat as well 🤔 


So TLDR is I think it would be ok to introduce the syntax in this PR, but then I would likely try and propose (as a follow on PR) some smarter configuration namespace resolution that didn't require explicitly typing out the namespace each time

Reply:
Sounds good. Let's get the "base" syntax in and then we can progressively optimize shortcuts for cases when there is no ambiguity.